### PR TITLE
Merge Bitpay's version of Bitcoin Core (additional indexes)

### PR DESCRIPTION
**Pull request by UdjinM6**
- Resolved conflicts:
.travis.yml _(ignoring changes, keeping dash version)_
src/Makefile.am
src/main.cpp
src/main.h
src/rpcclient.cpp
src/rpcserver.cpp
src/script/script.cpp
src/script/script.h

- Fixed tests:
qa/rpc-tests/addressindex.py
qa/rpc-tests/spentindex.py
qa/rpc-tests/txtindex.py
src/test/script_P2PKH_tests.cpp

### CODE REVIEW

**Comment by schinzelh** [comment]
Maybe we need a compiler switch (like DISABLE_WALLET) to guard these features

File: src/rpcserver.cpp [hunk 1]
```diff
@@ -292,8 +293,7 @@ static const CRPCCommand vRPCCommands[] =
     { "blockchain",         "verifytxoutproof",       &verifytxoutproof,       true  },
     { "blockchain",         "gettxoutsetinfo",        &gettxoutsetinfo,        true  },
     { "blockchain",         "verifychain",            &verifychain,            true  },
-    { "blockchain",         "invalidateblock",        &invalidateblock,        true  },
-    { "blockchain",         "reconsiderblock",        &reconsiderblock,        true  },
```
Comment (schinzelh) [inline src/rpcserver.cpp]: I guess you don't wont to remove these

Comment (UdjinM6) [inline src/rpcserver.cpp]: We somehow missed https://github.com/bitcoin/bitcoin/commit/5ebe0956b391cfa782fd97bb417d20b479df6a36 so I resolved it this way.

**Comment by schinzelh** [comment]
Is this still WIP?

**Comment by UdjinM6** [comment]
Changed title ;)

This merge is kind of huge so I still would like to hear any comments and/or (ut/n)ACKs from @evan82 though.

**Comment by schinzelh** [comment]
Yep, i agree - waiting for comments and/or (ut/n)ACKs from @evan82 :)

**Comment by evan82** [comment]
This is insanely useful and built entirely separate from core... merged :+1:

**Pull request merged**